### PR TITLE
Add selection function for comparison test

### PR DESCRIPTION
**Pull request by djbarnwal**
Adds Selection Comparison function `vlComparisonTest` 

vega/vega-lite#5068 uses this function.

### CODE REVIEW

**Comment by jheer** [comment]
Thanks for the PR! Adding @arvind to review prior to merge.

**Comment by jheer** [comment]
I pushed some changes to clean up the code and also removed the use of the Function constructor. Compiling a new function for every invocation is problematic from both a performance and a security perspective.

File: packages/vega-functions/src/codegen.js [hunk 1]
```diff
@@ -274,6 +275,7 @@ expressionFunction('treeAncestors', treeAncestors, dataVisitor);
 
 // register Vega-Lite selection functions
 expressionFunction('vlSelectionTest', selectionTest, selectionVisitor);
+expressionFunction('vlComparisonTest', comparisonTest, selectionVisitor);
```
Comment (jheer) [inline packages/vega-functions/src/codegen.js]: The name `comparisonTest` is rather generic, and breaks the convention of naming selection-specific methods with the `selection` prefix. Can we name it something like `selectionCompare` instead? The registered expression function name would then be `vlSelectionCompare`.

**Comment by arvind** [comment]
Stale and superseded by #3675

**Pull request closed without merging**